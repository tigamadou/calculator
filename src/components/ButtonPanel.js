/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const buttonGroups = [
  ['AC', '+/-', '%', 'รท'],
  ['7', '8', '9', 'x'],
  ['4', '5', '6', '-'],
  ['1', '2', '3', '+'],
  ['0', '.', '='],
];

const ButtonPanel = ({ handleClick }) => (
  <div className="panel">
    {buttonGroups.map((group, index) => (

      <div className="row" key={`row-${(index + 1)}`}>
        {group.map((button, yindex) => (
          (
            (yindex === (group.length - 1))
              ? <Button name={button} handleClick={handleClick} key={button} />
              : <Button name={button} handleClick={handleClick} key={button} color wide={button === '0'} />)

        ))}

      </div>
    ))}

  </div>
);

ButtonPanel.propTypes = {
  handleClick: PropTypes.func,
};

ButtonPanel.defaultProps = {
  handleClick: null,
};

export default ButtonPanel;
