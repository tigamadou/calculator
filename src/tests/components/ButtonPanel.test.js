import React from 'react';
import { render, screen } from '@testing-library/react';
import ButtonPanel from '../../components/ButtonPanel';

describe('ButtonPanel', () => {
  test('renders ButtonPanel component ', () => {
    render(<ButtonPanel />);
    // eslint-disable-next-line no-unused-expressions
    expect(screen.getAllByRole('button')).toBeInTheDocument;
  });
});
