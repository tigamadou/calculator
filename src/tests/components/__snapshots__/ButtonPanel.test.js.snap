// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel component tests It match with snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
